src/store/store.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useReducer, useEffect } from 'react';
+import { Todo } from '../types/Todo';
+import { useLocalStorage } from '../hooks/useLocalStorage';
+import { Dispatchers } from '../types/enums/Dispatchers';
+
+type Actions = { type: Dispatchers.Add; payload: Todo }
+| { type: Dispatchers.UpdateTitle; payload: Todo }
+| { type: Dispatchers.ChangeStatus; payload: number }
+| { type: Dispatchers.ChangeAllStatuses; payload: boolean }
+| { type: Dispatchers.DeleteWithId; payload: number }
+| { type: Dispatchers.DeleteComplited };
+
+const reducer = (state: Todo[], action: Actions) => {
+  switch (action.type) {
+    case Dispatchers.Add:
+      return [
+        ...state,
+        action.payload,
+      ];
+
+    case Dispatchers.DeleteComplited:
+      return [...state].filter(elem => !elem.completed);
```

Comment:
todo would more appropriate name of the item in this case instead of elem

Reply:
Done